src/docfx/lib/path/PathString.cs

```diff
@@ -35,13 +49,83 @@ namespace Microsoft.Docs.Build
 
         public static implicit operator string(PathString value) => value.Value;
 
+        /// <summary>
+        /// Concat two <see cref="PathString"/>s together.
+        /// </summary>
+        public static PathString operator +(PathString a, PathString b)
+        {
+            if (string.IsNullOrEmpty(a._value))
+                return b;
+
+            if (string.IsNullOrEmpty(b._value))
+                return a;
+
+            if (b._value[0] == '/')
+                return b;
+
+            var str = a._value[a._value.Length - 1] == '/'
+                ? a._value + b._value
+                : a.Value + '/' + b._value;
+
+            if (b._value[0] == '.')
+                return new PathString { _value = PathUtility.Normalize(str) };
+
+            return new PathString { _value = str };
+        }
+
+        /// <summary>
+        /// Check if the file is the same as matcher or is inside the directory specified by matcher.
+        /// </summary>
+        public bool StartsWithPath(PathString basePath, out PathString remainingPath)
+        {
+            var basePathValue = basePath.Value;
+            var pathValue = Value;
+
+            if (basePathValue.Length == 0)
+            {
+                remainingPath = this;
+                return true;
+            }
+
+            if (!pathValue.StartsWith(basePathValue, PathUtility.PathComparison))
+            {
+                remainingPath = default;
+                return false;
+            }
+
+            var i = basePathValue.Length;
+            if (basePathValue[i - 1] == '/')
+            {
+                // a/b starts with a/
+                remainingPath = new PathString { _value = pathValue.Substring(i) };
+                return true;
+            }
+
+            if (pathValue.Length <= i)
+            {
+                // a starts with a
+                remainingPath = default;
+                return true;
+            }
+
+            if (pathValue[i] == '/')
+            {
+                // a/b starts with a
+                remainingPath = new PathString { _value = pathValue.Substring(i + 1) };
+                return true;
+            }
+
+            remainingPath = default;
+            return false;
+        }
+
         private class PathStringJsonConverter : JsonConverter
         {
             public override bool CanConvert(Type objectType) => objectType == typeof(PathString);
 
             public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
             {
-                var value = reader.ReadAsString();
+                var value = serializer.Deserialize<string>(reader);
```

Comment:
what is the difference compared to the original?

Reply:
The original one throws when `PathString` is a dictionary value, because `ReadAsString`  throws when the current JsonReader is reading a property. `serializer.Deserialize<T>` resolves that.